test/unit/math/src/math.cpp

```diff
@@ -224,30 +225,9 @@ TEST_CASE("mathOps", "[math] [operator]")
      *     - call alpaka::meta::forEachType with the tuple in ForEachFunctor
      */
 
-    using TestAccs = alpaka::test::acc::EnabledAccs<
-        alpaka::dim::DimInt< 1u >,
-        std::size_t
-    >;
-    const unsigned long seed = 1337;
-    std::cout << "using seed: " << seed << "\n\n";
-    std::cout << "testing:\n "
-        << std::tuple_size<TestAccs>::value
-        << " - accelerators !\n"
-        << std::tuple_size<alpaka::test::unit::math::UnaryFunctors>::value
-        << " - unary math operators\n"
-        << std::tuple_size<alpaka::test::unit::math::BinaryFunctors>::value
-        << " - binary math operators\n"
-        << "testing with two data types\n"
-        << "total 2 * accelerators * (unary + binary) * capacity\n\n";
-
-
-    alpaka::meta::forEachType< TestAccs >(
-        ForEachFunctor< double >( ),
-        seed
-    );
-
-    alpaka::meta::forEachType< TestAccs >(
-        ForEachFunctor< float >( ),
-        seed
-    );
+    using Acc = std::tuple_element_t<0u, TestType>;
+    using Functor = std::tuple_element_t<1u, TestType>;
+
+    alpaka::meta::forEachType< DataTypes >(
+        TestTemplate< Acc, Functor >());
```

Comment:
It is possible to use:
```
alpaka::meta::CartesianProduct<std::tuple, TestAccs, DataTypes, Functors>
```
but then nvcc complains about: `error #456: excessive recursion at instantiation of class`.
Therefore we still need one indirection through forEachType`.